trl/trainer/judges.py

```diff
@@ -239,7 +238,11 @@ def judge(
             ranks[flip_mask] = ranks[flip_mask][:, ::-1]
 
         # Return the ranks or score probability
-        return softmax(ranks, axis=-1)[:, 0].tolist() if return_scores else ranks[:, 0].tolist()
+        if return_scores:
+            probs = np.exp(ranks) / sum(np.exp(ranks))
```

Comment:
I think we should use the shifted variant to avoid numerical instability: https://github.com/scipy/scipy/blob/92d2a8592782ee19a1161d0bf3fc2241ba78bb63/scipy/special/_logsumexp.py#L234

Reply:
done in https://github.com/huggingface/trl/pull/2298#discussion_r1824164828